### PR TITLE
test: cover task preprocessor workflow resolution

### PR DESCRIPTION
**Pull request by Peyton-Spencer**
## Summary

Adds deterministic task preprocessor coverage for workflow path resolution edge cases:

- rejects whitespace-only `preprocess_script` values through `runTaskPreprocessor`
- exercises default `groups/<folder>/task-workflows` resolution when no `workflowsDir` override is provided

## Test Count

Before:

- `bun test`: 2,382 pass, 0 fail, 6,362 assertions across 104 files
- `bun test --coverage`: 2,382 pass, 0 fail; all-files coverage 92.08% functions / 91.31% lines

After:

- `bun test`: 2,384 pass, 0 fail, 6,364 assertions across 104 files
- `bun test --coverage`: 2,384 pass, 0 fail; all-files coverage 92.18% functions / 91.34% lines

## Files Covered

- `src/task-preprocessor.test.ts`
- `src/task-preprocessor.ts`: improved to 100.00% functions / 99.20% lines

## Verification

- `bun test src/task-preprocessor.test.ts`
- `bun test`
- `bun test --coverage`

## Remaining Coverage Gaps

- `src/channels/discord.ts`, `src/channels/telegram.ts`, and `src/channels/whatsapp.ts` remain low-coverage integration-heavy adapters.
- `src/backends/local-backend.ts` still has large untested runtime/container orchestration branches.
- `src/index.ts` remains mostly uncovered because it combines startup orchestration, message routing, and side-effectful channel/runtime wiring.

## Summary by CodeRabbit

* **Tests**
  * Added test coverage for preprocessor validation (rejection of empty values) and default workflow directory resolution.

### CODE REVIEW

File: src/task-preprocessor.test.ts [hunk 1]
```diff
@@ -310,3 +311,38 @@ it('does nothing when no preprocessor is configured', () => {
...
+    error: 'preprocess_script must be a non-empty path',
+  });
+});
+
+it('resolves the default group task-workflows directory when no override is provided', () => {
+  const groupFolder = `preprocessor-test-${process.pid}`;
+  const groupWorkflowsDir = path.join(
+    GROUPS_DIR,
+    groupFolder,
+    TASK_WORKFLOWS_DIR,
```
Comment (Peyton-Spencer) [inline src/task-preprocessor.test.ts]: This fixture assumes `TASK_WORKFLOWS_DIR` is always relative to the group directory, but `defaultWorkflowsDir()` supports an absolute shared workflow directory and returns it directly. I reproduced this by running `TASK_WORKFLOWS_DIR=/tmp/omniclaw-abs-workflows-pr888 bun test src/task-preprocessor.test.ts`: the test writes under `groups/<group>/.../tmp/.../workflow.ts`, while `runTaskPreprocessor()` tries `/tmp/omniclaw-abs-workflows-pr888/workflow.ts`, so the test fails. Please either compute the fixture directory with the same absolute-vs-relative behavior as the implementation, or isolate the env before importing config.

**Review by Peyton-Spencer — COMMENTED** [review]
Reviewed the test-only change. Local checks passed under the default environment: `bun test src/task-preprocessor.test.ts`, `bun run typecheck`, and full `bun test` (2384 tests). I left one inline note about an env-specific fixture issue when `TASK_WORKFLOWS_DIR` is configured as an absolute path, which the implementation supports.

File: src/task-preprocessor.test.ts (hunk 1 above)
Comment (Peyton-Spencer) [reply src/task-preprocessor.test.ts]: Fixed in 204b649 by deriving the fixture directory with the same absolute-vs-relative `TASK_WORKFLOWS_DIR` behavior as `defaultWorkflowsDir()`. Verified with `bun test src/task-preprocessor.test.ts`, `TASK_WORKFLOWS_DIR=/tmp/omniclaw-abs-workflows-pr888 bun test src/task-preprocessor.test.ts`, `bun run format:check`, and `bun run typecheck`.
